x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureopenai/embeddings/AzureOpenAiEmbeddingsServiceSettings.java

```diff
@@ -80,19 +84,34 @@ private static CommonFields fromMap(
                         ServiceUtils.invalidSettingError(DIMENSIONS_SET_BY_USER, ModelConfigurations.SERVICE_SETTINGS)
                     );
                 }
+
+                // the similarity should never be in the request as Azure does not expose this, and we always default to DOT_PRODUCT
```

Comment:
Actually I think we do want to expose this to users when they configure the inference endpoint via the PUT request. The similarity field here will control what we place in the mapping for the dense vector field within elasticsearch: https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html

So we want users to be able to control that through what they set here. The similarity isn't something that we'll send to azure though like you mentioned.

Reply:
Ah - that makes sense - will fix this up.